Bottom: 5bc624dfae24567a90874ee97b8bcbd632d12ebb
Top:    693186532c793b5866ef9cac88a51bc76bac7cb6
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 15:00:57 +0800

autofs-5.0.7 - fix incorrect check in flag_is_owned()

The flag file code isn't used any more but this is clearly incorrect
so fix it in case it gets used sometime in the future.


---

diff --git a/daemon/flag.c b/daemon/flag.c
index f8fe163..db9a4bd 100644
--- a/daemon/flag.c
+++ b/daemon/flag.c
@@ -66,12 +66,11 @@ static int flag_is_owned(int fd)
 
 			continue;
 		}
-
-		/* Stale flagfile */
-		if (!tries)
-			return 0;
 	}
 
+	/* Stale flagfile */
+	if (!tries)
+		return 0;
 
 	if (pid) {
 		int ret;
